ksql-rest-app/src/test/resources/ksql-plan-schema/schema-final.json

```diff
@@ -0,0 +1,885 @@
+      "required" : [ "@type", "properties", "source", "formats" ]
+    },
+    "StreamMapValues" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "streamMapValuesV1" ],
+          "default" : "streamMapValuesV1"
+        },
+        "properties" : {
+          "oneOf" : [ {
+            "$ref" : "#/definitions/DefaultExecutionStepProperties"
+          } ]
+        },
+        "source" : {
+          "$ref" : "#/definitions/ExecutionStep"
+        },
+        "selectExpressions" : {
+          "type" : "array",
+          "items" : {
+            "type" : "string"
+          }
+        },
+        "selectNodeName" : {
+          "type" : "string"
+        }
+      },
+      "title" : "streamMapValuesV1",
+      "required" : [ "@type", "properties", "source", "selectExpressions", "selectNodeName" ]
+    },
+    "StreamSelectKey" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "streamSelectKeyV1" ],
+          "default" : "streamSelectKeyV1"
+        },
+        "properties" : {
+          "oneOf" : [ {
+            "$ref" : "#/definitions/DefaultExecutionStepProperties"
+          } ]
+        },
+        "source" : {
+          "$ref" : "#/definitions/ExecutionStep"
+        },
+        "fieldName" : {
+          "type" : "string"
+        },
+        "updateRowKey" : {
+          "type" : "boolean"
+        }
+      },
+      "title" : "streamSelectKeyV1",
+      "required" : [ "@type", "properties", "source", "fieldName", "updateRowKey" ]
+    },
+    "StreamSink" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "streamSinkV1" ],
+          "default" : "streamSinkV1"
+        },
+        "properties" : {
+          "oneOf" : [ {
+            "$ref" : "#/definitions/DefaultExecutionStepProperties"
+          } ]
+        },
+        "source" : {
+          "$ref" : "#/definitions/ExecutionStep"
+        },
+        "formats" : {
+          "$ref" : "#/definitions/Formats"
+        },
+        "topicName" : {
+          "type" : "string"
+        }
+      },
+      "title" : "streamSinkV1",
+      "required" : [ "@type", "properties", "source", "formats", "topicName" ]
+    },
+    "StreamSource" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "streamSourceV1" ],
+          "default" : "streamSourceV1"
+        },
+        "properties" : {
+          "oneOf" : [ {
+            "$ref" : "#/definitions/DefaultExecutionStepProperties"
+          } ]
+        },
+        "offsetReset" : {
+          "type" : "string",
+          "enum" : [ "EARLIEST", "LATEST" ]
+        },
```

Comment:
I don't think this should be in here.  CSAS/CTAS should _always_ be starting from the start of the steam IMHO.

Reply:
I agree for CTAS. CSAS I'm not so sure. It seems reasonable to start streaming from the end of a topic. The topic backing a stream probably has a limited retention policy, so you're not guaranteed any starting point anyway.

Reply:
CSAS should always start from the beginning.  If/when we chose to allow users to set other starting points it should be via SQL syntax, e.g. the much discussed `WHERE ROWTIME <op> <exp>` or `WHERE (partition, offset) <op> latest()` etc.   

NOW is the time to remove stuff like this IMHO